### PR TITLE
Add XPU profiler correctness test suite

### PR DESCRIPTION
**Pull request by PawelSwider2000**
Fixes: [#2844](https://github.com/intel/torch-xpu-ops/issues/2844)

Introduce test/profiling/test_profiler_correctness.py, a proper
unittest-based test suite that validates XPU profiler output
correctness.  Tests are converted from assertion-free reproducer
scripts and extended with explicit assertions.

Covered scenarios:
- Mixed XPU ops produce CPU and XPU events with valid chrome trace
- H2D transfers and compute ops are captured and correctly typed
- GPU kernel times are non-zero (regression test for missing XPU
  kernel-time bug on MaxUnpool2d fwd + bwd)
- Repeated profiler invocations produce consistent kernel sets across
  iterations (no intermittent missing events)
- torch.compile'd functions on XPU are profiled correctly
- All trace events have non-negative duration and timestamp
- Chrome trace JSON structure conforms to the chrome-tracing spec
- Custom record_function labels appear in profiler output
- torch.profiler.schedule fires the trace callback the expected number
  of times
- profile_memory=True produces [memory] instant events in the trace
- Kernel events carry an args dict enabling op correlation checks
- Profiler can be started and stopped repeatedly without errors
- Backward-pass XPU kernels are captured alongside forward ops
- Explicit H2D and D2H transfers appear in profiled events

Wire the new test file into CI via linux-uttest action.
Adjust test_for_overlapping_kernels.py for compatibility.
Run test_xpu_profiler.py tests

### CODE REVIEW

File: .github/actions/linux-uttest/action.yml [hunk 1]
```diff
@@ -114,19 +114,12 @@ runs:
...
-          tee ${{ github.workspace }}/ut_log/xpu_profiling/issue_reproduce/profile_partial_runtime_ops.log
-        python -u test/profiling/triton_xpu_ops_time.py | \
-          tee ${{ github.workspace }}/ut_log/xpu_profiling/issue_reproduce/triton_xpu_ops_time.log
+        
+        python -m pytest -s test/profiling/test_profiler_correctness.py | \
+          tee ${{ github.workspace }}/ut_log/xpu_profiling/test_profiler_correctness.log
+        # Overlapping kernels test (depends on rn50.log.json from RN50 run above)
         python -m pytest -s test/profiling/test_for_overlapping_kernels.py | \
-          tee ${{ github.workspace }}/ut_log/xpu_profiling/issue_reproduce/test_for_overlapping_kernels.log
+          tee ${{ github.workspace }}/ut_log/xpu_profiling/test_for_overlapping_kernels.log
```
[the remark this answers is not in the document]
Comment (PawelSwider2000) [reply .github/actions/linux-uttest/action.yml]: @chuanqi129 It is a real issue here? As we were using `pytest ... | tee ...` before...

File: test/profiling/test_profiler_correctness.py [hunk 2]
```diff
@@ -0,0 +1,598 @@
...
+from torch.profiler import DeviceType
+from torch.testing._internal.common_utils import run_tests, TestCase
+from torch.utils._triton import has_triton
+
+
+class TestProfilerCorrectness(TestCase):
+    ACTIVITIES = [
+        torch.profiler.ProfilerActivity.CPU,
+        torch.profiler.ProfilerActivity.XPU,
+    ]
```
[the remark this answers is not in the document]
Comment (PawelSwider2000) [reply test/profiling/test_profiler_correctness.py:26]: This tests are meant to be run only for XPU

File: test/profiling/test_profiler_correctness.py [hunk 3]
```diff
@@ -0,0 +1,598 @@
...
+
+        # Chrome trace must contain kernel category events
+        self._gen_and_check_json(prof, expected_cats=["kernel"])
+
+    def test_time_precision(self):
+        x = torch.randn(3, 3, device="cpu")
+
+        # Warm up
+        self.compute(x)
+
```
[the remark this answers is not in the document]
Comment (PawelSwider2000) [reply test/profiling/test_profiler_correctness.py]: Done

File: test/profiling/test_profiler_correctness.py [hunk 4]
```diff
@@ -0,0 +1,598 @@
...
+                0,
+                f"Event '{event.name}' has negative cpu_time_total: "
+                f"{event.cpu_time_total}",
+            )
+            self.assertGreaterEqual(
+                event.self_cpu_time_total,
+                0,
+                f"Event '{event.name}' has negative self_cpu_time_total: "
+                f"{event.self_cpu_time_total}",
+            )
```
[the remark this answers is not in the document]
Comment (PawelSwider2000) [reply test/profiling/test_profiler_correctness.py:352]: Done

File: .github/actions/linux-uttest/action.yml [hunk 5]
```diff
@@ -114,19 +114,12 @@ runs:
...
-        python -u test/profiling/correlation_id_mixed.py | \
-          tee ${{ github.workspace }}/ut_log/xpu_profiling/issue_reproduce/correlation_id_mixed.log
-        python -u test/profiling/reproducer.missing.gpu.kernel.time.py | \
-          tee ${{ github.workspace }}/ut_log/xpu_profiling/issue_reproduce/reproducer.missing.gpu.kernel.time.log
-        python -u test/profiling/time_precision_in_profile.py | \
-          tee ${{ github.workspace }}/ut_log/xpu_profiling/issue_reproduce/time_precision_in_profile.log
-        python -u test/profiling/profile_partial_runtime_ops.py | \
-          tee ${{ github.workspace }}/ut_log/xpu_profiling/issue_reproduce/profile_partial_runtime_ops.log
-        python -u test/profiling/triton_xpu_ops_time.py | \
-          tee ${{ github.workspace }}/ut_log/xpu_profiling/issue_reproduce/triton_xpu_ops_time.log
```
Comment (RUIJIEZHONG66166) [inline .github/actions/linux-uttest/action.yml:127]: Please do not remove previous profile UT test. These tests can serve as a reference to help verify the results of the new UT test. You can just add new test. Others are LGTM.

Comment (PawelSwider2000) [reply .github/actions/linux-uttest/action.yml:127]: Done, previous scripts are restored.

**Comment by PawelSwider2000** [comment]
@RUIJIEZHONG66166 On demand tests: https://github.com/intel/torch-xpu-ops/actions/runs/22665659896

**Review by RUIJIEZHONG66166 — APPROVED** [review]

**Comment by PawelSwider2000** [comment]
xpu-profiling suite passed, errors observed on CI are unrelated
